pkg/identity/cache/allocator.go

```diff
@@ -205,13 +233,14 @@ func AllocateIdentity(ctx context.Context, owner IdentityAllocatorOwner, lbls la
 		return reservedIdentity, false, nil
 	}
 
-	if !identity.RequiresGlobalIdentity(lbls) && localIdentities != nil {
+	if !identity.RequiresGlobalIdentity(lbls) {
+		<-localIdentityAllocatorInitialized
 		return localIdentities.lookupOrCreate(lbls)
 	}
 
 	// This will block until the kvstore can be accessed and all identities
 	// were successfully synced
-	WaitForInitialIdentities(ctx)
+	WaitForInitialGlobalIdentities(ctx)
```

Comment:
`WaitForInitialGlobalIdentities()` returns an error, we should check it (and that should handle the `IdentityAllocator == nil` case below, if I follow correctly).